spec/services/decidim/action_delegator/participants_csv_importer_spec.rb

```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+
+require "spec_helper"
+
+describe Decidim::ActionDelegator::ParticipantsCsvImporter do
+  let(:current_user) { create(:user, :confirmed, :admin, organization: organization) }
+  let(:organization) { create(:organization) }
+  let(:consultation) { create(:consultation, organization: organization) }
+  let(:current_setting) { create(:setting, consultation: consultation) }
+  let(:valid_csv_file) { File.open("spec/fixtures/valid_participants.csv") }
+  let(:invalid_csv_file) { File.open("spec/fixtures/invalid_participants.csv") }
+
+  describe "#import!" do
+    context "when the rows in the csv file are valid" do
+      subject { described_class.new(valid_csv_file, current_user, current_setting) }
+
+      it "Import all rows from csv file" do
+        expect do
+          subject.import!
+        end.to change(Decidim::ActionDelegator::Participant, :count).by(4)
+      end
+
+      it "returns a summary of the import" do
+        import_summary = subject.import!
+
+        expect(import_summary[:error_rows]).to eq []
+        expect(import_summary[:imported_rows]).to eq 4
+        expect(import_summary[:total_rows]).to eq 4
+      end
+    end
+
+    context "when the rows in the csv file are not valid" do
+      subject { described_class.new(invalid_csv_file, current_user, current_setting) }
+
+      it "creates participants from valid rows" do
+        expect do
+          subject.import!
+        end.to change(Decidim::ActionDelegator::Participant, :count).by(2)
+      end
+
+      it "returns a summary of the import" do
+        import_summary = subject.import!
+
+        expect(import_summary[:error_rows].pluck(:row_number)).to eq [1, 4]
+        expect(import_summary[:imported_rows]).to eq 2
+        expect(import_summary[:total_rows]).to eq 5
+      end
+    end
+
+    context "when participant with this email already exists" do
+      subject { described_class.new(valid_csv_file, current_user, current_setting) }
+
+      let!(:participant) { create(:participant, email: "baz@example.org", setting: current_setting) }
+
+      it "does not create a new participant" do
+        expect do
+          subject.import!
+        end.to change(Decidim::ActionDelegator::Participant, :count).by(3)
+      end
+    end
+
+    context "when participant exists with another data" do
+      subject { described_class.new(valid_csv_file, current_user, current_setting) }
+
+      let!(:participant) { create(:participant, phone: "123456789", setting: current_setting) }
+
+      it "does not change the data of the existing participant" do
+        expect do
+          subject.import!
+        end.not_to change { participant.reload.phone }.from("123456789")
+      end
+    end
+  end
```

Comment:
I think we need to create more cases here,
for instance I'm missing the case on importing a CSV without emails (in case the verifier is in "phone" only mode).
A csv like
```csv
phone,weight
6660000,1.5
6660001,1
6660002,2
```

Also, let's try cases where not all columns are present (for instance the phone or the weight) and the import works equally well (as long as the setting is properly configured)